app/controllers/planned_runs_controller.rb

```diff
@@ -54,9 +57,24 @@ def update
 
     #Update all runs if Planned_run is changed
     Run.where(planned_run_id: @planned_run.id).find_each do |run|
-      run.date = @planned_run.date
-      run.training_plan = @planned_run.training_plan
-      run.save
+      update = true
+      if params[:checks]
+        params[:checks].each do |checks|
```

Comment:
Couldn't this be made into a one liner with a select?   update = params[:checks].first(|check| the test here)  ????